### PR TITLE
Fix load balancing loss

### PR DESCRIPTION
**Pull request by crewtool**
# Description
load balancing losses from different gradient checkpointing steps were inferencing. Clearing losses fixes problem.
Additionaly  `total_cross_entropy_loss`  consitsted of loss with additional losses.

This change needs https://github.com/llm-random/llm-random/pull/331 to be merged first.

# Checklist
   - [ ] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [ ] Are all variables set with logical defaults that are backward compatible?
   - [ ] I attached link to neptune if it was necessary
   - [ ] Have you successfully executed the linter and tests?
   - [ ] Are all functions concise and don't require splitting?
   - [ ] Is there documentation for arguments and complex logic?
   - [ ] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [ ] Are all functions placed in appropriate files?
   - [ ] Is the PR name meaningful and descriptive?
   - [ ] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [ ] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/utils/model_utils.py [hunk 1]
```diff
@@ -365,6 +365,7 @@ def retrieve_additional_losses(model: torch.nn.Module):
         load_balancing_losses = torch.stack(load_balancing_losses)
         load_balancing_loss = torch.mean(load_balancing_losses)
         losses["load_balancing_loss"] = load_balancing_loss
+        model.forward_pass_cache["load_balancing_losses"] = []
```
Comment (Simontwice) [inline research/conditional/utils/model_utils.py:368]: we clear the fw pass cache after every step. What is the purpose of this?

Comment (Simontwice) [reply research/conditional/utils/model_utils.py:368]: ah, I see - gradient accumulation. It has to be empty for the tensors connected to aux loss to be freed after the optimize step

**Comment by maciek-pioro** [comment]
@crewtool is this still relevant?

**Comment by crewtool** [comment]
Looks like it became Szymons branch for running exps.
